<launch>   
    <include file="$(find hakuroukun_tasp_with_cones)/launch/gazebo_cones.launch" />

    <arg name="left_laser_filter_file" default="left_laser_filter.yaml"/>
    <arg name="right_laser_filter_file" default="right_laser_filter.yaml"/>
    <node pkg="laser_filters" type="scan_to_scan_filter_chain" name="left_laser_filter">
        <rosparam command="load" file="$(find hakuroukun_launch)/config/$(arg left_laser_filter_file)" />
        <remap from="scan" to="/left_lidar/scan" />
        <remap from="scan_filtered" to="/left_lidar/scan_filtered" />
    </node>
    <node pkg="laser_filters" type="scan_to_scan_filter_chain" name="right_laser_filter">
        <rosparam command="load" file="$(find hakuroukun_launch)/config/$(arg right_laser_filter_file)" />
        <remap from="scan" to="/right_lidar/scan" />
        <remap from="scan_filtered" to="/right_lidar/scan_filtered" />
    </node>

    <node pkg="ira_laser_tools" name="laserscan_multi_merger" type="laserscan_multi_merger" output="screen" required="true" >
        <param name="destination_frame" value="laser_link"/>
        <param name="cloud_destination_topic" value="/merged_cloud"/>
        <param name="scan_destination_topic" value="/scan_multi"/>
        <param name="laserscan_topics" value ="/left_lidar/scan_filtered /right_lidar/scan_filtered" /> 
        <param name="angle_min" value="-2.61"/>
        <param name="angle_max" value="2.61"/>
        <param name="angle_increment" value="0.034906"/>
        <param name="scan_time" value="0.066666"/> 
        <param name="range_min" value="1.0"/>
        <param name="range_max" value="20.0"/>
    </node>

    <node name="hakuroukun_pose" pkg="hakuroukun_pose" type="hakuroukun_pose_node" output="screen" required="true" >
        <param name="~log" value="false"/>
        <param name="~log_folder" value="$(find hakuroukun_pose)/log_data"/>
        <param name="~publish_rate" value="0.1"/>
        <param name="~rotation_angle" value="-90"/> 
        <param name="~gps_to_rear_axis" value="0.6"/>
        <param name="~imu_mode" value="quaternion"/>
        <param name="~imu_epsilon" value="0.05"/>
        <param name="~imu_calibration_threshold" value="10"/>
    </node>
</launch>